geode-core/src/main/java/org/apache/geode/internal/monitoring/executor/AbstractExecutor.java

```diff
@@ -51,9 +56,25 @@ protected AbstractExecutor(String groupName, long threadID) {
 
   public void handleExpiry(long stuckTime, Map<Long, ThreadInfo> threadInfoMap) {
     incNumIterationsStuck();
+    sendAlertForThreadStuckForLong(stuckTime, threadInfoMap);
     logger.warn(createThreadReport(stuckTime, threadInfoMap));
   }
 
+  private void sendAlertForThreadStuckForLong(long stuckTime, Map<Long, ThreadInfo> threadInfoMap) {
+    if (maxThreadStuckTime <= 0) {
+      return;
+    }
+    if (threadInfoMap.get(threadID) == null) {
+      return;
+    }
+    if (stuckForGood) {
+      return;
+    }
+    if (stuckTime > maxThreadStuckTime) {
+      stuckForGood = true;
+      logger.fatal(createThreadReport(stuckTime, threadInfoMap));
```

Comment:
`fatal` should be reserved for logging about catastrophic failure that may result in data corruption. The server should probably be taken off-line or bounced.

`error` should be used to indicate that something failed but there should be no data corruption and it should generally be ok to allow the server to continue running.

I think you probably want this log statement to be at `error` level which will still generate an `alert`.

I'd like to add @dschneider-pivotal as a reviewer to confirm the above. Darrel, if you want to give me good definitions, I'd be happy to add it to the Apache Geode Wiki.

Reply:
@kirklund I think the alert will not be generated by default if `error` is used. In order to see the alert when using `error`, the `alertLevel` in the `DistributedMXBean` should be changed from `severe` to a lower level.

See https://geode.apache.org/docs/guide/114/managing/management/notification_federation_and_alerts.html

Anyhow, I agree with you that it makes sense to use `error` given that having a thread stuck is not necessarily a catastrophic failure that may result in data corruption.

Reply:
@kirklund The problem I see with using `error` in this case is that, as you need to change the `alertLevel` in the `DistributedMXBean` to a lower level, the number of notifications received will be much higher, something that you do not necessarily want, for example because more resources will be used.

There are other alerts in Geode issued when a fatal error is sent that do not necessarily mean a catastrophic failure, for example if the `ack-severe-threshold` is surpassed and that feature is enabled.

Given that this is also a configurable feature, whoever makes use of it should know what to expect and how to act (according to the documentation) when the alert is received so I still think it makes sense to use `severe` to make sure the alert is always received if the feature is enabled.